llama-index-core/llama_index/core/base/embeddings/base_sparse.py

```diff
@@ -0,0 +1,381 @@
+"""Base embeddings file."""
+
+import asyncio
+import math
+from abc import abstractmethod
+from collections import defaultdict
+from typing import Any, Callable, Coroutine, Dict, List, Optional
+
+from llama_index.core.bridge.pydantic import (
+    BaseModel,
+    Field,
+    ConfigDict,
+)
+from llama_index.core.constants import DEFAULT_EMBED_BATCH_SIZE
+from llama_index.core.instrumentation import DispatcherSpanMixin
+from llama_index.core.instrumentation.events.embedding import (
+    SparseEmbeddingEndEvent,
+    SparseEmbeddingStartEvent,
+)
+import llama_index.core.instrumentation as instrument
+from llama_index.core.utils import get_tqdm_iterable
+from llama_index.core.async_utils import run_jobs
+
+
+dispatcher = instrument.get_dispatcher(__name__)
+
+SparseEmbedding = Dict[int, float]
+
+
+def sparse_similarity(
+    embedding1: SparseEmbedding,
+    embedding2: SparseEmbedding,
+) -> float:
+    """Get embedding similarity."""
```

Comment:
Does it make sense to have a similarity mode here as we do with dense embeddings?

And I guess we didn't want to add another dep in order to compute the similarities and norms?

I've used scipy sparse (https://docs.scipy.org/doc/scipy/reference/sparse.html#) in past for sparse linalg before.

Reply:
It's such a simple calculation, I felt like python is fast enough lol 

We could maybe use numpy.

I was going to add a model, but said screw it, since it's not even configurable for dense embeddings 😅